homeassistant/components/media_player/directv.py

```diff
@@ -118,13 +162,39 @@ def media_content_id(self):
             return None
         return self._current['programId']
 
+    @property
+    def media_content_type(self):
+        """Return the content type of current playing media."""
+        if 'episodeTitle' in self._current:
+            return MEDIA_TYPE_TVSHOW
+        return MEDIA_TYPE_MOVIE
+
     @property
     def media_duration(self):
         """Return the duration of current playing media in seconds."""
         if self._is_standby:
             return None
         return self._current['duration']
 
+    @property
+    def media_position(self):
+        """Position of current playing media in seconds."""
+        if self._is_standby:
+            return None
+
+        return self._lastposition
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid.
+
+        Returns value from homeassistant.util.dt.utcnow().
+        """
+        if self._is_standby:
+            return None
+
+        return self._lastupdate
```

Comment:
This looks like it will spam the state machine even if position hasn't been updated.

Reply:
@MartinHjelmare Thus any updates to attributes (and not just state) will result in the state machine having to do something? 

I also got a different question and I checked the dev docs but could not specifically see it. There are attributes and they are returned by device_state_attributes. But then there are others which are not returned by device_state_attributes (i.e. the media_position_updated_at). What is the difference or best approach for attributes?

Thanks for your insights!

Reply:
Most properties are returned as part of `state_attributes`, and will create a new state in the state machine when updated. `device_state_attributes` are merged with `state_attributes` in the base entity class before reporting the state to the state machine.

https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/media_player/__init__.py#L785-L790